File: src/main/java/eu/mshade/mwork/event/EventContext.java
package eu.mshade.mwork.event;

public interface EventContext<E> {

    EventContext<E> withEventFilter(EventFilter eventFilter);

    EventContext<E> withEventPriority(EventPriority eventPriority);

    EventPriority getEventPriority();

    EventFilter getEventFilter();

    EventListener<E> getEventListener();

    Class<E> getEventType();
}

File: src/main/java/eu/mshade/mwork/event/EventContext.kt
package eu.mshade.mwork.event

interface EventContext<E> {

    var eventPriority: EventPriority
    var eventFilter: EventFilter
    var eventType: Class<E>
    var eventListener: EventListener<E>

    fun withEventFilter(eventFilter: EventFilter) : EventContext<E>;
    fun withEventPriority(eventPriority: EventPriority) : EventContext<E>;
}